### DISCUSSION TITLE
Semicolon URI parameter pair separator

### DISCUSSION THREAD

**Discussion by pangyre** [mojolicious/mojo#1944] [Q&A]
I hope this isn't a foolish question or that I've missed a baseline change in standards but I personally prefer the semicolon to the ampersand for URI query strings. I find it easier to read, to type, and especially like to see them in HTML in place of &amp;amp;s.

I'm a mojo n00b but it appears only the ampersand is supported. Am I wrong? Is there a reason? Would a patch to include the semicolon as an acceptable optional delimiter be accepted? Are there implications in URI construction there? As with the need to set it globally like $URI::DEFAULT_QUERY_FORM_DELIMITER in URI:: space.

Thank you!

**Comment by Grinnz** [comment]
semicolons are not supported as the web has rarely ever used them in URLs in the past 20 years. with proper HTML escaping no one should ever notice the escaped form. I would not expect a patch to be accepted.

**Reply by pangyre** [reply]
I understand they are uncommon but it is a standard, supported by URI and others, and the W3C recommends using them. Perl, to me, is about sane affordances. I included a properly escaped ampersand to show I know how to do it, I just don’t like it, and as an example of why they are visually unpleasant in source HTML.

**Reply by kraih** [reply]
Please refer to the specific specs (and sections) you care about. Because the specs we currently use do not recommend it.
